frontend/catalyst/qfunc.py

```diff
@@ -124,7 +124,7 @@ def __call__(self, *args, **kwargs):
 
         # TODO: Move the capability loading and validation to the device constructor when the
         # support for old device api is dropped.
-        program_features = ProgramFeatures(self.device.shots is not None)
+        program_features = ProgramFeatures(shots_present=bool(self.device.shots))
```

Comment:
This change is made quite a few times throughout this PR (mostly in tests). If you pass `shots=None` to the new device API, it gets wrapped in a `Shots` object, and `Shots(None) is not None` is `True` instead of `False`. This means that devices with the new API and `shots=None` are being given `shots_present=True` and incorrectly being treated as finite shot devices in `qjit_capabilites`. 

If we cast it to a boolean instead, we get the intended behaviour for both the old and new device APIs.

Reply:
Oh thanks for catching this! A change that slipped through the cracks I suppose. Verification showing its usefulness yet again :D